vassal-app/src/main/java/VASSAL/build/module/Chatter.java

```diff
@@ -686,9 +725,539 @@ public static void main(String[] args) {
+  /**
+   * @deprecated use {@link GlobalOptions#getPlayerId()}
+   */
   @Deprecated(since = "2020-08-06", forRemoval = true)
   public String getHandle() {
     return GlobalOptions.getInstance().getPlayerId();
   }
+
+
+  static class ChatTransferHandler extends TransferHandler {
+
+    protected Transferable createTransferable(JComponent c) {
+      final JEditorPane pane = (JEditorPane) c;
+      final String htmlText = pane.getText();
+      final String plainText = extractText(new StringReader(htmlText));
+      return new ChatTransferable(plainText, htmlText);
+    }
+
+    public String extractText(Reader reader) {
+      final ArrayList<String> list = new ArrayList<String>();
+
+      final HTMLEditorKit.ParserCallback parserCallback = new HTMLEditorKit.ParserCallback() {
+        public void handleText(final char[] data, final int pos) {
+          list.add(new String(data));
+        }
+
+        public void handleStartTag(HTML.Tag tag, MutableAttributeSet attribute, int pos) {
+        }
+
+        public void handleEndTag(HTML.Tag t, final int pos) {
+        }
+
+        public void handleSimpleTag(HTML.Tag t, MutableAttributeSet a, final int pos) {
+          if (t.equals(HTML.Tag.BR)) {
+            list.add("\n");
+          }
+        }
+
+        public void handleComment(final char[] data, final int pos) {
+        }
+
+        public void handleError(final String errMsg, final int pos) {
+        }
+      };
+      try {
+        new ParserDelegator().parse(reader, parserCallback, true);
+      }
+      catch (IOException e) {
+        e.printStackTrace();
+      }
+      String result = "";
+      for (final String s : list) {
+        result += s;
+      }
+      return result;
+    }
+
+    @Override
+    public void exportToClipboard(JComponent comp, Clipboard clip, int action) throws IllegalStateException {
+      if (action == COPY) {
+        clip.setContents(this.createTransferable(comp), null);
+      }
+    }
+
+    @Override
+    public int getSourceActions(JComponent c) {
+      return COPY;
+    }
+  }
+
+  static class ChatTransferable implements Transferable {
+    private final DataFlavor[] supportedFlavors;
+
+    private final String plainData;
+    private final String htmlData;
+
+    public ChatTransferable(String plainData, String htmlData) {
+      this.plainData = plainData;
+      this.htmlData = htmlData;
+
+      try {
+        supportedFlavors = new DataFlavor[]{
+          new DataFlavor("text/html;class=java.lang.String"),
+          new DataFlavor("text/plain;class=java.lang.String")
+        };
+      }
+      catch (ClassNotFoundException e) {
+        throw new ExceptionInInitializerError(e);
+      }
+    }
+
+    public DataFlavor[] getTransferDataFlavors() {
+      return supportedFlavors;
+    }
+
+    public boolean isDataFlavorSupported(DataFlavor flavor) {
+      for (final DataFlavor supportedFlavor : supportedFlavors) {
+        if (supportedFlavor == flavor) {
```

Comment:
Should this be `supportedFlavor.equals(flavor)` or are you really intending to compare by identity?